src/OrbitGl/TimelineUi.cpp

```diff
@@ -24,11 +25,13 @@ void TimelineUi::RenderLines(PrimitiveAssembler& primitive_assembler, uint64_t m
        timeline_ticks_.GetAllTicks(min_timestamp_ns, max_timestamp_ns)) {
     float world_x = timeline_info_interface_->GetWorldFromUs(
         tick_ns / static_cast<double>(kNanosecondsPerMicrosecond));
-    int screen_x = viewport_->WorldToScreen(Vec2(world_x, 0))[0];
-    primitive_assembler.AddVerticalLine(
-        Vec2(screen_x, GetPos()[1]), GetHeightWithoutMargin(), GlCanvas::kZValueTimeBar,
-        tick_type == TimelineTicks::TickType::kMajorTick ? kTimelineMajorTickColor
-                                                         : kTimelineMinorTickColor);
+    if (IsElementOf(world_x, ClosedInterval::FromValues(GetPos()[0], GetPos()[0] + GetSize()[0]))) {
```

Comment:
Nit: I prefer 
```
if (auto visible_range = ClosedInterval::FromValues(GetPos()[0], GetPos()[0] + GetSize()[0]); 
    IsElementOf(world_x, visible_range) {
....
}
```
to make this more readable.

Not sure whether `visible_range` is a good name.

Reply:
Nice idea! go/totw/165. I will use this from now on

Reply:
I finally moved to the beginning since it's better to calculate once instead of the loop but I used Vicky's trick at the bottom!